crates/circuit_encodings/src/lib.rs

```diff
@@ -222,10 +245,9 @@ impl<
             previous_head: self.head, // unchanged
             previous_tail: old_tail,
             num_items: self.num_items,
-            round_function_execution_pairs: states,
         };
 
-        (old_tail, intermediate_info)
+        (round_states, intermediate_info)
```

Comment:
Change here looks suspicious. Round states where previously in the info, and first output was just a previous tail before push

Reply:
Previous tail has never been used anywhere, so i replaced it with `round_states` to avoid saving them in data structures